src/idl_gen_csharp.cpp

```diff
@@ -1225,7 +1225,7 @@ class CSharpGenerator : public BaseGenerator {
             code += "Add";
             code += GenMethod(vector_type);
             code += "(";
-            code += SourceCastBasic(vector_type);
+            code += SourceCastBasic(vector_type, false);
```

Comment:
At the moment there is no support of arrays with optional enums, so isOptional = false

Reply:
Maybe this should be a comment in the code